src/helpers/channels/create_channel.ts

```diff
@@ -17,14 +17,22 @@ import { calculateBits } from "../../util/permissions.ts";
 /** Create a channel in your server. Bot needs MANAGE_CHANNEL permissions in the server. */
 export async function createChannel(
   guildId: string,
-  options?: CreateGuildChannel,
+  options?: CreateGuildChannel
 ) {
   const requiredPerms: Set<PermissionStrings> = new Set(["MANAGE_CHANNELS"]);
 
+  //Integration with permissions for channel cloning
```

Comment:
I'm not so a fan of this solution maybe discuss more on discord. My preference would be that the cloneChannel function converts the overwrites to a compatible version.

Reply:
Agreed!

Reply:
This was my original idea, I just wasn't sure how to do it. I don't really like how I did this either. Definitely open to change here.

Reply:
You can loop over the overwrites and just use the `calculateBits()` function to change the allow and deny props.

Reply:
Changed it to use `calculatePermissions()` in cloneChannel and removed the integration in createChannel.